unix/vncserver/vncsession.c

```diff
@@ -345,13 +345,54 @@ switch_user(const char *username, uid_t uid, gid_t gid)
     }
 }
 
+static void
+mkvncdir(const char *dir)
+{
+    if (mkdir(dir, 0755) == -1) {
+        if (errno != EEXIST) {
+            syslog(LOG_CRIT, "Failure creating \"%s\": %s", dir, strerror(errno));
+            _exit(EX_OSERR);
+        }
+
+#ifdef HAVE_SELINUX
+        int result;
+        if (selinux_file_context_verify(dir, 0) == 0) {
+            result = selinux_restorecon(dir, SELINUX_RESTORECON_RECURSE);
+
+            if (result < 0) {
+                syslog(LOG_WARNING, "Failure restoring SELinux context for \"%s\": %s", dir, strerror(errno));
+            }
+        }
+#endif
```

Comment:
This should not be needed and indicates something is wrong elsewhere with the SELinux policy.

Reply:
So sorry. I didn't realize that this was moved code. I thought it was new stuff. It's better kept as it was.

It is only needed for the legacy dir, though, since it is there because we changed the type of `~/.vnc`. Ideally, we only use it for that, but I guess that will be messy?

Could you add a comment that this is only needed to handle historical type changes for the legacy dir?